### PR TITLE
Add script to programmatically generate syntax

### PR DESCRIPTION
**Pull request by giordano**
NOTE: at this moment, this is only a proof-of-concept, but it will serve as
basis for generating the syntax with PyOracc

### CODE REVIEW

File: python/generate_syntax_highlightining.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+support = {}
+protocols = '|'.join([s.lower() for s in AtfLexer.protocols])
+protocol_keywords = '|'.join([s.lower() for s in AtfLexer.protocol_keywords])
+dollar_keywords = '|'.join([s.lower() for s in AtfLexer.dollar_keywords])
+support["patterns"] = [{"name": "support.class.hash.atf",
+                        # Interlinear translations (`tr`) need to be treaded
+                        # specially, because they're the only protocol allowed
+                        # to be followed by something else (the specification
+                        # of the language `.<LANG>`) before the colon.
+                        "match": f"^#(tr\\.[[:alpha:]]+|{protocols})?\\b: *({protocol_keywords})?"},
```
[the remark this answers is not in the document]
Comment (giordano) [reply python/generate_syntax_highlightining.py:34]: Good point, not sure why `?` is there at all, probably a leftover of some experiments.

`\s` [matches](https://macromates.com/manual/en/regular_expressions#syntax_oniguruma) _any_ whitespace character, including new lines `\n` and `\r`, I'm pretty sure we don't want them.  Maybe `[ \t]` is a safer choice?

File: python/generate_syntax_highlightining.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+# Initialise the dictionary
+data = {}
+
+data["$schema"] = "https://raw.githubusercontent.com/martinring/tmlanguage/master/tmlanguage.json"
+data["name"] = "ATF"
+data["patterns"] = [{"include": "#keywords"}, {"include": "#support"}, {"include": "#strings"}]
+
+keywords = {}
+structures = '|'.join([s.lower() for s in AtfLexer.structures])
```
[the remark this answers is not in the document]
Comment (giordano) [reply python/generate_syntax_highlightining.py:16]: Right, thanks!

File: python/generate_syntax_highlightining.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ageorgou) [reply python/generate_syntax_highlightining.py:34]: Right, of course! Space or `\t` should be enough, yes.

File: python/generate_syntax_highlightining.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+     # We highlight the entire line starting with `$`, but we capture the known
+     # keywords.
+     "begin": "^\\$",
+     "end": "\\n",
+     "patterns": [
+         {
+             # At the moment we don't do anything special about these keywords,
+             # they're highlighted like the rest of the line, but we can
+             # capture them and give them a different name.
+             "name": "support.variable.dollar-keyword.atf",
```
Comment (ageorgou) [inline python/generate_syntax_highlightining.py:62]: If we did want to highlight these in a different colour, would we need to give them a semantically-different name?

File: python/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+Generate a virtual environment with, for example,
+
+```
+$ mkvirtualenv syntax
```
Comment (ageorgou) [inline python/README.md:4]: Now that virtual environments have been in the standard library for a long time (don't tell Jython!), let's use that


```suggestion
```bash
$ python -m venv syntax
```

File: python/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+where `syntax` is the name we're giving to the environment.  After creating it,
+you can activate this environment with
+
+```
+$ workon syntax
```
Comment (ageorgou) [inline python/README.md:11]: `venv` workflow:

```suggestion
```bash
$ source syntax/bin/activate
```

**Review by ageorgou — APPROVED** [review]
Looks reasonable! Only minor nitpicks in the instructions.

File: python/generate_syntax_highlightining.py (hunk 3 above)
Comment (giordano) [reply python/generate_syntax_highlightining.py:62]: That's my understanding as well

File: python/README.md (hunk 4 above)
Comment (giordano) [reply python/README.md]: Well, I think I copied the instructions from Nammu :sweat_smile:

**Pull request merged**